### ISSUE TITLE
Error "Division by zero" for some JPEG files

### ISSUE THREAD

**Issue by GitWidi** [nextcloud/maps#443]
For some image files I recognized that the following error occurs in the protocol:
```
[PHP] Error: Division by zero at /var/www/html/custom_apps/maps/lib/Service/PhotofilesService.php#407 at 2020-08-28T08:30:00+00:00
```

When analysing my image files I noticed that there are incorrect GPS information in the EXIF data. 
```
GPSLongitude=[ 0/0 , 0/0 , 0/0 ]
GPSLatitude=[ 0/0 , 0/0 , 0/0 ]
```
Yes, this issue has been discussed before (#361) and I am aware that these are incorrect GPS information. But it is not in my possibilities to avoid or correct this error. Therefore, such files will appear again and again. 

Is it possible to handle this special case so that these unnecessary error messages no longer appear?

I also noticed that the app "metadata" recognizes that the GPS coordinates are useless. There this special case is already handled in the PHP program.  (see `Metadata\Service\MetadataService::evalGpsCoord()`).

Thanks and regards
Friedbert

**Comment by DeMyrax** [comment]
Same Issue here, how to Avoid this errors? My logs are full of it.

Thanks

**Comment by UweKrause** [comment]
Just for reference:

Division by zero happens here:
https://github.com/nextcloud/maps/blob/3e600cbee501a897c6ba2f7e99aafb7272b7fba6/lib/Service/PhotofilesService.php#L413

Function to detect malformed exif-data before calculating stuff:
https://github.com/nextcloud/maps/blob/3e600cbee501a897c6ba2f7e99aafb7272b7fba6/lib/Service/PhotofilesService.php#L354

The Code @GitWidi mentioned is probably this: (?)
https://github.com/gino0631/nextcloud-metadata/blob/5188cffcd7b635c93d42162a80f034fdfeea5972/lib/Service/MetadataService.php#L598

At the moment I try to reproduce the bug and find the not catched gps exif data that triggers it.

**Comment by UweKrause** [comment]
I would add another check, BEFORE https://github.com/nextcloud/maps/blob/3e600cbee501a897c6ba2f7e99aafb7272b7fba6/lib/Service/PhotofilesService.php#L361

Exif metadata is user controlled data and therefore it can be anything.
I would do a strict check, like this:

```php
// check if all parts are in a correct format
if(!($this->isValidFormatLatOrLong($exif["GPSLatitude"][0])
    && $this->isValidFormatLatOrLong($exif["GPSLatitude"][1])
    && $this->isValidFormatLatOrLong($exif["GPSLatitude"][2])
    && $this->isValidFormatLatOrLong($exif["GPSLongitude"][0])
    && $this->isValidFormatLatOrLong($exif["GPSLongitude"][1])
    && $this->isValidFormatLatOrLong($exif["GPSLongitude"][2]))
    ) {
    return false;
}
```

```php
/* helper function, checks valid format
 * Latitude or longitude is valid if
 * - is numeric
 * - or is a valid fraction, where the denominator is not 0
 */
private function isValidFormatLatOrLong($latOrLong) {
    
    if (is_numeric($latOrLong)) {
        return true;
    }
    
    if (str_contains($latOrLong, "/")) {
        $e = explode("/", $latOrLong, 2);
        return is_numeric($e[0]) && is_numeric($e[1]) && $e[1]!=0;        
    }
    
    return false;
}
```

What do you think?
**What else would be valid Lat/Long Format?**

**Comment by UweKrause** [comment]
just btw: the code I suggested works for me (very small instance with 2 users).
I patched it manually into the server and got rid of the error messages.

**But**: I am not aware of the usual formats the lat/long field is supposed to have.
I am not sure if the `is_numeric()` check even makes sense.
I have no time for the next weeks to dive into the lat/long and/or jpeg specifications.
And I don't know what steps are needed to add code to this repository.
That are the reasons I did not provide a PR so far.

**If anyone wants to take care of this, please feel free to use (or feel free to not use!) the code snipped provided.**

**Comment by tacruc** [comment]
I think this is fixed in v0.2.5 feel free to reopen if not.
